src/payment/strategies/credit-card-redirect/credit-card-redirect-payment-strategy.ts

```diff
@@ -0,0 +1,65 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { some } from 'lodash';
+
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { RequestError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+export default class CreditCardRedirectPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _formPoster: FormPoster
+    ) {}
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentData = payment && payment.paymentData;
+
+        if (!payment || !paymentData) {
```

Comment:
You already evaluated payment above

```suggestion
        if (!paymentData) {
```

Reply:
linter throws error on line 32 if I don't validate `payment`

Reply:
Yeah, that's true.